app/code/Magento/AdvancedSearch/Model/ResourceModel/Index.php

```diff
@@ -150,6 +152,63 @@ public function getPriceIndexData($productIds, $storeId)
         return $priceProductsIndexData[$websiteId];
     }
 
+    /**
+     * Return array of inventory data products
+     *
+     * @param null|array $productIds
+     * @return array
+     * @since 100.1.0
+     */
+    protected function _getCatalogProductInventoryData($productIds = null)
+    {
+        $connection = $this->getConnection();
+        $catalogProductIndexInventorySelect = [];
+
+        foreach ($this->dimensionCollectionFactory->create() as $dimensions) {
+            if (!isset($dimensions[WebsiteDimensionProvider::DIMENSION_NAME]) ||
+                $this->websiteId === null ||
+                $dimensions[WebsiteDimensionProvider::DIMENSION_NAME]->getValue() === $this->websiteId) {
+                $select = $connection->select()->from(
+                    $this->tableResolver->resolve('cataloginventory_stock_status', $dimensions),
+                    ['product_id', 'website_id', 'stock_status']
+                );
+                if ($productIds) {
+                    $select->where('product_id IN (?)', $productIds);
+                }
+                $catalogProductIndexInventorySelect[] = $select;
+            }
+        }
+
+        $catalogProductIndexInventoryUnionSelect = $connection->select()->union($catalogProductIndexInventorySelect);
+
+        $result = [];
+        foreach ($connection->fetchAll($catalogProductIndexInventoryUnionSelect) as $row) {
+            $result[$row['website_id']][$row['product_id']] = (int) $row['stock_status'];
+        }
+
+        return $result;
+    }
+
+    /**
+     * Retrieve inventory data for product
+     *
+     * @param null|array $productIds
+     * @param int $websiteId
+     * @return array
+     */
+    public function getInventoryIndexData($productIds, $websiteId = null)
```

Comment:
introduce the public function make the Semantic test failed. @sidolov Can you give us some advice about that ?

Reply:
Backward capability don't allow introduce new public methods to class marked as `@api` and should be replaced by new class